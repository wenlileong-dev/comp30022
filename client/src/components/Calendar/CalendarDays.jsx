import React from "react";

import CalendarDay from "./CalendarDay";

function CalendarDays(props) {
  let firstDay = new Date(props.year, props.month, 1);
  const getDaysInMonth = (date) => {
    return new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
  };
  let daysInMonth = getDaysInMonth(new Date(props.year, props.month));

  return (
    <div className="calendar-container">
      {[...Array(firstDay.getDay())].map((day, index) => {
        return <CalendarDay day="" key={`a${index}`} />;
      })}
      {[...Array(daysInMonth)].map((day, index) => {
        return (
          <CalendarDay
            day={index + 1}
            month={props.month}
            year={props.year}
            event={props.events[index]}
            key={index}
          />
        );
      })}
    </div>
  );
}

export default CalendarDays;
