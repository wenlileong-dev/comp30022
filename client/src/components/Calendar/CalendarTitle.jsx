import React, { useState } from "react";
import ArrowLeftIcon from "@mui/icons-material/ArrowLeft";
import ArrowRightIcon from "@mui/icons-material/ArrowRight";
import AddBoxIcon from "@mui/icons-material/AddBox";
import Grid from "@mui/material/Grid";
import Button from "@mui/material/Button";

import CalendarPopup from "./CalendarPopup";

function CalendarTitle(props) {
  const monthNames = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
  ];

  let [isPopupOpen, setIsPopupOpen] = useState(false);

  // function togglePopup(event) {
  //   console.log(event.target);
  //   setIsPopupOpen(!isPopupOpen);
  // }

  function handleClose(event) {
    event.stopPropagation();
    setIsPopupOpen(false);
  }
  function toggleAddEvent() {
    setIsPopupOpen(true);
  }

  return (
    <React.Fragment>
      <div className="calendar-title">
        <Grid container spacing={3}>
          <Grid item xs={12} sm={9}>
            <Grid container spacing={2}>
              <Grid item xs={3} className="calendar-month">
                <ArrowLeftIcon onClick={props.prevMonth}></ArrowLeftIcon>
              </Grid>
              <Grid item xs={6}>
                <span className="calendar-month">
                  {monthNames[props.month]} {props.year}
                </span>
              </Grid>
              <Grid item xs={3} className="calendar-month">
                <ArrowRightIcon onClick={props.nextMonth}></ArrowRightIcon>
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={12} sm={3}>
            <Button
              variant="contained"
              color="primary"
              onClick={toggleAddEvent}
              id="add-event-button"
              startIcon={<AddBoxIcon />}
            >
              New Event
            </Button>
          </Grid>
        </Grid>
      </div>
      {isPopupOpen && (
        <CalendarPopup
          renderType="new-event"
          handleClose={handleClose}
          isPopupOpen={isPopupOpen}
        ></CalendarPopup>
      )}
    </React.Fragment>
  );
}
export default CalendarTitle;
