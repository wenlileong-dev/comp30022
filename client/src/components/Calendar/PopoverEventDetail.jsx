import React from "react";
import Grid from "@material-ui/core/Grid";
import PopoverEventComponent from "./PopoverEventComponent";

function PopoverEventDetail(props) {
  return (
    <React.Fragment>
      <Grid container spacing={3}>
        {props.events &&
          props.events.map((event, index) => {
            return (
              <PopoverEventComponent
                event={event}
                setEvent={props.setEventDetail}
                openEditEvent={props.toggleEditEvent}
                key={index}
              />
            );
          })}
      </Grid>
    </React.Fragment>
  );
}

export default PopoverEventDetail;
