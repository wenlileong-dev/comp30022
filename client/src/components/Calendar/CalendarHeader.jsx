import React from "react";
import "./Calendar.css";
function CalendarHeader() {
  return (
    <div className="calendar-header">
      <div>Sunday</div>
      <div>Monday</div>
      <div>Tuesday</div>
      <div>Wednesday</div>
      <div>Thursday</div>
      <div>Friday</div>
      <div>Saturday</div>
    </div>
  );
}

export default CalendarHeader;
