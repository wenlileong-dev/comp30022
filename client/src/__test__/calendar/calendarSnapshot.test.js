import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import CalendarTitle from "../../components/Calendar/CalendarTitle";
import CalendarHeader from "../../components/Calendar/CalendarHeader";
import CalendarDays from "../../components/Calendar/CalendarDays";
import PopoverEventDetail from "../../components/Calendar/PopoverEventDetail";
const mockEvents = [
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [],
  [
    {
      people: [],
      meetingNotes: "",
      _id: "6155452462f2dd3e2ce8abda",
      title: "Test 24",
      description: "Description",
      date: "2021-09-24T05:03:21.000Z",
      time: "2021-09-30T07:00:00.018Z",
      eventType: "Online",
      location: "Zoom",
      userID: "614d7378de23b63284235ab6",
      timeInNumbers: 1020,
      id: "6155452462f2dd3e2ce8abda",
    },
  ],
  [
    {
      people: ["Anna", "Joe"],
      meetingNotes: "",
      _id: "6155456262f2dd3e2ce8abdd",
      title: "Test 25",
      date: "2021-09-25T05:04:19.000Z",
      time: "2021-09-30T08:00:00.823Z",
      eventType: "Online",
      location: "Zoom",
      userID: "614d7378de23b63284235ab6",
      timeInNumbers: 1080,
      id: "6155456262f2dd3e2ce8abdd",
    },
  ],
  [],
  [
    {
      people: [],
      meetingNotes: "",
      _id: "6155494c24739e1bd8d4b9a9",
      title: "Test 27",
      date: "2021-09-27T05:21:05.000Z",
      time: "2021-09-30T10:01:00.461Z",
      eventType: "Online",
      location: "Google Meet",
      userID: "614d7378de23b63284235ab6",
      timeInNumbers: 1201,
      id: "6155494c24739e1bd8d4b9a9",
    },
  ],
  [
    {
      people: [],
      meetingNotes: "",
      _id: "6155495724739e1bd8d4b9ac",
      title: "Test 28",
      date: "2021-09-28T05:21:18.000Z",
      time: "2021-09-30T11:00:00.332Z",
      eventType: "Online",
      location: "Microsoft Team",
      userID: "614d7378de23b63284235ab6",
      timeInNumbers: 1260,
      id: "6155495724739e1bd8d4b9ac",
    },
  ],
  [],
  [
    {
      people: [],
      meetingNotes: "",
      _id: "6155457a62f2dd3e2ce8abe0",
      title: "Test 30",
      date: "2021-09-30T05:04:48.000Z",
      time: "2021-09-30T13:00:00.859Z",
      eventType: "Online",
      location: "Zoom",
      userID: "614d7378de23b63284235ab6",
      timeInNumbers: 1380,
      id: "6155457a62f2dd3e2ce8abe0",
    },
  ],
];
const month = 8;
const year = 2021;
test("Snaphot Testing for Calendar Title", () => {
  const component = renderer.create(
    <CalendarTitle month={month} year={year} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Snapshot Testing for Calendar Header", () => {
  const component = renderer.create(<CalendarHeader />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Snapshot Testing for Calendar Days", () => {
  const component = renderer.create(
    <CalendarDays month={month} year={year} events={mockEvents} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Snaphot Testing for Event Details", () => {
  const component = renderer.create(
    <PopoverEventDetail events={mockEvents[29]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
