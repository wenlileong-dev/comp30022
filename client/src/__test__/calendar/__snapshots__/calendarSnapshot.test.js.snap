// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snaphot Testing for Calendar Title 1`] = `
<div
  className="calendar-title"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9 css-d75gfe-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 calendar-month css-j23p4t-MuiGrid-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14 7-5 5 5 5V7z"
            />
          </svg>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-ayy8wd-MuiGrid-root"
        >
          <span
            className="calendar-month"
            title="calendar-title"
          >
            September
             
            2021
          </span>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 calendar-month css-j23p4t-MuiGrid-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m10 17 5-5-5-5v10z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3 css-10wfvsf-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="add-event-button"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
        </span>
        New Event
      </button>
    </div>
  </div>
</div>
`;

exports[`Snaphot Testing for Event Details 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1402ce4-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-l4g9cp-MuiGrid-root"
      data-testid="event-day-component"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        style={
          Object {
            "backgroundColor": "#EAEEF3",
          }
        }
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            className="MuiTypography-root MuiTypography-h5 css-1asgzld-MuiTypography-root"
          >
            11:00 PM
             
            Test 30
          </div>
          <p
            className="MuiTypography-root MuiTypography-body1 css-pqpbz4-MuiTypography-root"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-xor7ac-MuiTypography-root"
          />
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-tdqki2-MuiTypography-root"
          >
            Online
             via 
            Zoom
          </h6>
          <div
            className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
          >
            <button
              className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-1k23hlb-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              id="space-btw-event-button"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Open Event
            </button>
            <button
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root css-11qr2p8-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="EditIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              Edit Event
            </button>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-3ogxn6-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot Testing for Calendar Days 1`] = `
<div
  className="calendar-container"
>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      1
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      2
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      3
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      4
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      5
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      6
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      7
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      8
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      9
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      10
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      11
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      12
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      13
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      14
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      15
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      16
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      17
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      18
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      19
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      20
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      21
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      22
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      23
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      24
    </p>
    <div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
      >
        Test 24
      </p>
    </div>
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      25
    </p>
    <div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
      >
        Test 25
      </p>
    </div>
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      26
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      27
    </p>
    <div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
      >
        Test 27
      </p>
    </div>
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      28
    </p>
    <div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
      >
        Test 28
      </p>
    </div>
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      29
    </p>
    <div />
  </div>
  <div
    className="calendar-day"
    data-testid="calendar-a-day"
    onClick={[Function]}
  >
    <p>
      30
    </p>
    <div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
      >
        Test 30
      </p>
    </div>
  </div>
</div>
`;

exports[`Snapshot Testing for Calendar Header 1`] = `
<div
  className="calendar-header"
  data-testid="mytestid"
>
  <div>
    Sun
  </div>
  <div>
    Mon
  </div>
  <div>
    Tue
  </div>
  <div>
    Wed
  </div>
  <div>
    Thu
  </div>
  <div>
    Fri
  </div>
  <div>
    Sat
  </div>
</div>
`;
